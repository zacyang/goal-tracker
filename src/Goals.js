import React, {Component} from 'react';
class Goals extends Component{
  constructor(props) {
    super(props);
  }

  render() {
    return (
            <ul>
            {
                this.props.goals.map((item, index) =>
                (<li key={index}>{item}
                  <button onClick={(event)=>this.props.onRemove(index)}>X</button>
                </li>) )
            }
        </ul>
    );
  }
}

export default Goals;
