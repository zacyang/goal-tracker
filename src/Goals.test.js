import React from 'react';
import ReactDOM from 'react-dom';
import Goals from './Goals.js';
import renderer from 'react-test-renderer';

test('Default snapshot test', ()=> {
    const items = [];
   const component = renderer.create(<Goals goals={items}> </Goals>, );
   let tree = component.toJSON();

   expect(tree).toMatchSnapshot();
});