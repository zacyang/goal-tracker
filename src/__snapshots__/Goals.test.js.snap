// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gaols snapshot test 1`] = `
<ul
  className="list-group"
>
  <a
    className="list-group-item disabled"
    href="#"
  >
    current goals
  </a>
</ul>
`;
